astroquery/esa/jwst/tests/test_jwsttap.py

```diff
@@ -0,0 +1,1088 @@
+        parameters['verbose'] = verbose
+        parameters['dump_to_file'] = dump_to_file
+        parameters['background'] = background
+        parameters['upload_resource'] = upload_resource
+        parameters['upload_table_name'] = upload_table_name
+        tap.launch_job(query,
+                       name=name,
+                       output_file=output_file,
+                       output_format=output_format,
+                       verbose=verbose,
+                       dump_to_file=dump_to_file,
+                       background=background,
+                       upload_resource=upload_resource,
+                       upload_table_name=upload_table_name,
+                       async_job=True)
+        dummyTapHandler.check_call('launch_job_async', parameters)
+
+    def test_list_async_jobs(self):
+        dummyTapHandler = DummyTapHandler()
+        tap = JwstClass(dummyTapHandler)
+        # default parameters
+        parameters = {}
+        parameters['verbose'] = False
+        tap.list_async_jobs()
+        dummyTapHandler.check_call('list_async_jobs', parameters)
+        # test with parameters
+        dummyTapHandler.reset()
+        parameters['verbose'] = True
+        tap.list_async_jobs(verbose=True)
+        dummyTapHandler.check_call('list_async_jobs', parameters)
+
+    def test_query_region(self):
+        connHandler = DummyConnHandler()
+        tapplus = TapPlus("http://test:1111/tap", connhandler=connHandler)
+        tap = JwstClass(tapplus)
+        # Launch response: we use default response because the
+        # query contains decimals
+        responseLaunchJob = DummyResponse()
+        responseLaunchJob.set_status_code(200)
+        responseLaunchJob.set_message("OK")
+        jobDataFile = data_path('job_1.vot')
+        jobData = utils.read_file_content(jobDataFile)
+        responseLaunchJob.set_data(method='POST',
+                                   context=None,
+                                   body=jobData,
+                                   headers=None)
+        # The query contains decimals: force default response
+        connHandler.set_default_response(responseLaunchJob)
+        sc = SkyCoord(ra=29.0, dec=15.0, unit=(u.degree, u.degree),
+                      frame='icrs')
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc)
+            print(err.value.args[0])
+        assert "Missing required argument: 'width'" in err.value.args[0]
+
+        width = Quantity(12, u.deg)
+        height = Quantity(10, u.deg)
+
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width)
+        assert "Missing required argument: 'height'" in err.value.args[0]
+
+        # Test observation_id argument
+        with pytest.raises(ValueError) as err:
+            tap.query_region(sc, width=width, height=height, observation_id=1)
+        assert "observation_id must be string" in err.value.args[0]
```

Comment:
would you test it with a valid string input, too?

Reply:
Tested with valid inputs after all assertions